pandas/tests/copy_view/test_option_deprecation.py

```diff
@@ -0,0 +1,25 @@
+import importlib
+
+import pandas as pd
+import pandas._testing as tm
+
+
+def test_configs_deprecated():
+    # GH#57872
+    with tm.assert_produces_warning(FutureWarning, match="is deprecated"):
+        pd.options.mode.copy_on_write = True
+
+    with tm.assert_produces_warning(FutureWarning, match="is deprecated"):
+        pd.options.future.no_silent_downcasting = True
+
+
+def test_env_variable():
+    # GH#57872
+    import os
+
+    os.environ["PANDAS_COPY_ON_WRITE"] = "1"
```

Comment:
Could you use `monkeypatch.setenv`? https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=monkeypatch#pytest.MonkeyPatch.setenv